.github/workflows/integration-tests.yml

```diff
@@ -49,15 +57,19 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: '3.11'
-      - name: Install python deps
-        run: pip install -r ./tests/consensus_tests/requirements.txt
+      - name: Install dependencies
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+
+          poetry -C tests check --lock
+          poetry -C tests install --no-root
       - name: Build
         run: cargo build --features "service_debug data-consistency-check"
       - name: Run integration tests - 1 peer
-        run: ./tests/integration-tests.sh distributed
+        run: poetry -C tests run ./tests/integration-tests.sh distributed
         shell: bash
       - name: Run integration tests - multiple peers - pytest
-        run: pytest ./tests/consensus_tests
+        run: poetry -C tests run pytest tests/consensus_tests
```

Comment:
@generall These are different. But I assume you're fine with this because you can still invoke `pytest` directly locally, am I right?

Reply:
yes, exactly. And I can use my own env as well